File: solutions/lua/triangle/1/triangle.lua
local triangle = {}

function triangle.kind(a, b, c)
    assert(a > 0, 'Input Error')
    assert(b > 0, 'Input Error')
    assert(c > 0, 'Input Error')
    assert(a + b > c, 'Input Error')
    assert(a + c > b, 'Input Error')
    assert(b + c > a, 'Input Error')
   
    if a == b and a == c then
      return 'equilateral'
    end
    if a == b or a == c or b == c then
      return 'isosceles'
    end
    return 'scalene'
end

return triangle

File: solutions/scala/triangle/1/src/main/scala/Triangle.scala
case class Triangle(a: Double, b: Double, c: Double) {
  private def valid =
    a > 0 && b > 0 && c > 0 && a + b > c && a + c > b && b + c > a
  def equilateral = valid && a == b && b == c
  def isosceles = valid && (a == b || a == c || b == c)
  def scalene = valid && a != b && a != c && b != c
}
